jerry-core/ecma/builtin-objects/ecma-builtin-bigint.c

```diff
@@ -36,6 +52,289 @@
+    ecma_bigint_digit_t *result_number_p = ECMA_BIGINT_GET_DIGITS (result_p, 0);
+    int32_t first_cell = 0;
+    uint32_t surplus_bits =
+      (whole_part > 0) ? (uint32_t) (whole_part * size_of_divisor_in_bits) : (uint32_t) size_of_divisor_in_bits;
+    uint32_t mask_bit = (whole_part == 0) ? (uint32_t) input_bits : (uint32_t) input_bits - surplus_bits;
+
+    if (mask_bit == 0)
+    {
+      mask_bit = size_of_divisor_in_bits - 1;
+    }
+
+    uint32_t check_sign_mask = (uint32_t) 1 << (mask_bit - 1);
+    uint32_t mask = ((uint32_t) 1 << mask_bit) - 1;
+    uint32_t last_cell = (exact_size >= sizeof (uint32_t)) ? (uint32_t) (min_size / sizeof (uint32_t)) - 1 : 0;
+    bool is_positive = false;
+    bool is_representation_positive = false;
+
+    if (is_signed)
+    {
+      if (input_bigint_sign && ((~digits_p[last_cell] + 1) & check_sign_mask) == 0)
+      {
+        is_positive = true;
+      }
+
+      if ((digits_p[last_cell] & check_sign_mask) == 0)
+      {
+        is_representation_positive = true;
+      }
+    }
+
+    do
+    {
+      *result_number_p++ =
+        (is_representation_positive || (!is_signed && !input_bigint_sign)) ? *digits_p++ : ~(*digits_p++);
+      first_cell--;
+    } while (digits_p < digits_end_p);
+
+    int16_t equal_bits = 0;
+
+    if (remainder != 0)
+    {
+      equal_bits = -1;
+    }
+
+    int32_t last_cell_negative = (last_cell != 0) ? ((int32_t) last_cell * (-1)) : -1;
+    bool is_zero_values = false;
+
+    if (!is_signed)
+    {
+      if (input_bigint_sign)
+      {
+        is_zero_values = true;
+      }
+    }
+    else
+    {
+      if (((digits_p[-1] & check_sign_mask) > 0) || (result_number_p[-1] & check_sign_mask) > 0)
+      {
+        is_zero_values = true;
+      }
+    }
+
+    if (is_zero_values)
+    {
+      result_number_p[first_cell] += 1;
+
+      if (result_number_p[first_cell] == 0)
+      {
+        do
+        {
+          result_number_p[++first_cell] += 1;
+        } while (first_cell != equal_bits);
+
+        first_cell = last_cell_negative;
+      }
+    }
+
+    result_number_p[-1] &= mask;
+    uint32_t surplus = (uint32_t) (min_size - exact_size) / sizeof (ecma_char_t);
+    uint32_t new_size = result_p->u.bigint_sign_and_size;
+
+    if ((min_size - exact_size) % (sizeof (ecma_char_t)) > 0 && surplus == 0)
+    {
+      surplus += (uint32_t) sizeof (ecma_char_t);
+    }
+    else
+    {
+      surplus = (uint32_t) (surplus * sizeof (ecma_char_t));
+    }
+
+    if (min_size / JERRY_BITSINBYTE < 1)
+    {
+      surplus = 0;
+    }
+
+    if (is_signed)
+    {
+      if (result_p->u.bigint_sign_and_size > exact_size && min_size > sizeof (uint32_t)
+          && result_number_p[last_cell_negative] < 1)
+      {
+        new_size -= surplus;
+      }
+
+      new_size += 1;
+
+      if (is_positive || ((digits_p[-1] & check_sign_mask) == 0 && !input_bigint_sign))
+      {
+        new_size -= 1;
+      }
+    }
+
+    while (first_cell != 0)
```

Comment:
IMO, this can be simplified like this:
```
    while (first_cell != 0 && result_number_p[first_cell] != 0)
    {
      first_cell++;
    }
```

OR ` while (first_cell++ != 0 && result_number_p[first_cell] != 0) { }`

Reply:
You meant `result_number_p[first_cell] == 0` right?

Reply:
right, my bad.